lib/Slider.js

```diff
@@ -640,31 +630,40 @@ export class Slider extends ShapedAssetEntity {
      * Handles moving the handle when track is clicked
      */
     #handleClick() {
-		// Check if the draw command is issued after logic
-		// Prevents issues where slider is non-responsive
-		if (
-			this.#lastFrameDrawn !== this.#pen.frameCount - 1 &&
-			this.#lastFrameDrawn !== this.#pen.frameCount
-		) {
-			return;
-		}
-
 		if (!this.isHoverOnHandle() && this.isHovering()) {
 			if (this.#pen.mouse.leftReleased) {
 				this.#setClosestStepValue();
 			}
 		}
 	}
+
+    internalUpdate() {
+		this.#handleDrag();
+		this.#handleClick();
+		this.#handleKeyboardInput();
+
+		if (this.#dragStart || this.isHoverOnHandle() || this.isHovering()) {
+			this.#isAnimating = true;
+			this.#handleIsAnimating = true;
+			this.#focussed = true;
+		} else {
+			this.#focussed = false;
+			this.#animProgress = 0;
+			this.#handleAnimProgress = 0;
+		}
+	}
+
     /**
      * Draws the slider to the canvas based on its current state.
      */
     draw() {
 		if (this.exists === false) {
-            return null;
+            return;
         }
+        this.update();
```

Comment:
Let's get update out of draw. The Update() function should be what you are doing with internalUpdate().

Update shouldn't be something that user code can trigger, it's a step prior to the drawing and user code like you've done with internal update. That should solve a lot of your worries about it getting called multiple times in a frame too.

Reply:
Done.